### DISCUSSION TITLE
OME signed policy question, can’t get right signature to play ll-hls m3u8 url

### DISCUSSION THREAD

**Discussion by howard5248** [OvenMediaLabs/OvenMediaEngine#1256] [Help] [answered]
Hello,
I use **OME v0.15.11** to provide llhls stream and add the config in the Server.xml virtualhosts part, followed by OME [signedpolicy](https://airensoft.gitbook.io/ovenmediaengine/access-control/signedpolicy).
```
<SignedPolicy>
    <PolicyQueryKeyName>policy</PolicyQueryKeyName>
    <SignatureQueryKeyName>signature</SignatureQueryKeyName>
    <SecretKey>aKq#1kj</SecretKey>

    <Enables>
        <Publishers>hls,llhls</Publishers>
    </Enables>
</SignedPolicy>
```
And use the url to create a signed URL for the llhls.m3u8. The secret key is `aKq#1kj`. [signed_policy_url_generator.sh](https://github.com/AirenSoft/OvenMediaEngine/blob/release/misc/signed_policy_url_generator.sh)
The signed url `http://130.211.250.252/app/stream/llhls.m3u8?policy=eyJ1cmxfZXhwaXJlIjoxNjg1NDE5MjEyfQ&signature=oC-FGuhjGFgSGW1xGSub0GiwRbc`. But I still got 401 responses. I check the log, it said `Signature value is invalid(expected : YY4FjualjKcMsYg2sergT2N53I4 | input : oC-FGuhjGFgSGW1xGSub0GiwRbc)`. I wonder how to create the right signature for the signed URL to play the stream.
Thank you~

**Comment by dimiden** [comment, marked as answer]
@howard5248 
The signature should include the port.
If you are using the ```misc/signed_policy_url_generator.sh```, you can use the following command:

```
$ ./signed_policy_url_generator.sh 'aKq#1kj' http://130.211.250.252:80/app/stream/llhls.m3u8 signature policy '{"url_expire":1685419212}'
[URL] http://130.211.250.252:80/app/stream/llhls.m3u8?policy=eyJ1cmxfZXhwaXJlIjoxNjg1NDE5MjEyfQ&signature=YY4FjualjKcMsYg2sergT2N53I4
[Percent encoded URL] http%3A//130.211.250.252%3A80/app/stream/llhls.m3u8%3Fpolicy%3DeyJ1cmxfZXhwaXJlIjoxNjg1NDE5MjEyfQ%26signature%3DYY4FjualjKcMsYg2sergT2N53I4
```

**Reply by howard5248** [reply]
Thank you for your replied!
